WordPress/gutenberg
# Add block spacing and layout to Post Template

## What?


Fixes #44899 and  #44557.

* Converts the custom "grid" layout in the Post Template block to use the grid layout type from #49018.
* The grid is configured to be static with a user-defined number of columns and a identical responsive behaviour to the current custom styes.
* The controls to toggle between List and Grid layouts are moved from Query into Post Template toolbar.
* Adds block spacing support to Post Template.

Note: the deprecation had to be done at Query level, because that's where the legacy layout type is stored. This doesn't really work on the server side though, so I opted to leave legacy classnames and styles for unchanged blocks on the front end.

## Testing Instructions





1. Go to the Home template in a block theme and check that Query and Post Template display as expected on this branch: should look exactly the same as on trunk before any changes are made. 
2. Set the layout to Grid in Post Template toolbar, and change the number of columns in the sidebar. This should work the same as on trunk.
3. Change block spacing from the Post Template block sidebar; check that it works for both the list and the grid layout options.

### Testing Instructions for Keyboard


## Screenshots or screencast
